core/metadata/src/impl/LoadUnitMultiInstanceProtoHook.ts

```
@@ -1,25 +1,25 @@
 
 export class LoadUnitMultiInstanceProtoHook implements LifecycleHook<LoadUnitLifecycleContext, LoadUnit> {
-  multiInstanceClazzSet: Set<EggProtoImplClass> = new Set();
+  static multiInstanceClazzSet: Set<EggProtoImplClass> = new Set();
 
-  async preCreate(ctx: LoadUnitLifecycleContext, loadUnit: LoadUnit): Promise<void> {
-    const clazzList = ctx.loader.load();
-    const multiInstanceClazzList = Array.from(this.multiInstanceClazzSet);
+  static setAllClassList(clazzList: readonly EggProtoImplClass[]) {
     for (const clazz of clazzList) {
       if (PrototypeUtil.isEggMultiInstancePrototype(clazz)) {
         this.multiInstanceClazzSet.add(clazz);
```

Biome

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)